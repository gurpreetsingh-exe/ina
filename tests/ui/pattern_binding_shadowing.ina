// COMPILE_FAIL

using Unit::Unit;
type Unit = Unit

using E::*;
type E = 
    | UnitVariant
    | TupleVariant(i32)

type Empty = {}

fn main() {
    match Unit {
        Unit => {}
    }

    let e = Empty {};
    match e {
        Empty => {}
    }

    match 0 {
        TupleVariant => {},
    }
}
