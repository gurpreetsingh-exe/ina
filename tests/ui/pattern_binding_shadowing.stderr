error: bindings cannot shadow constructors
    ╭─[tests/ui/pattern_binding_shadowing.ina:20:9]
    │
 20 │         Empty => {}
    │         ──┬──
    │           ╰─── try specifying the pattern arguments: `Empty { .. }`
    ∙
────╯
error: bindings cannot shadow constructors
    ╭─[tests/ui/pattern_binding_shadowing.ina:24:9]
    │
 24 │         TupleVariant => {},
    │         ──────┬─────
    │               ╰─────── try specifying the pattern arguments: `TupleVariant(..)`
    ∙
────╯
error: mismatch types
    ╭─[tests/ui/pattern_binding_shadowing.ina:24:9]
    │
 24 │         TupleVariant => {},
    │         ──────┬─────
    │               ╰─────── expected `integer`, found `fn(i32) -> E`
    ∙
────╯
